packages/react-native-codegen/src/parsers/typescript/modules/index.js

```diff
@@ -315,41 +335,32 @@ function translateTypeAnnotation(
                     ),
                   );
 
-                if (propertyTypeAnnotation.type === 'FunctionTypeAnnotation') {
-                  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
-                    hasteModuleName,
-                    property.typeAnnotation.typeAnnotation,
-                    property.key,
-                    propertyTypeAnnotation.type,
-                  );
-                }
-
-                if (propertyTypeAnnotation.type === 'VoidTypeAnnotation') {
-                  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
-                    hasteModuleName,
-                    property.typeAnnotation.typeAnnotation,
-                    property.key,
-                    'void',
-                  );
-                }
-
-                if (propertyTypeAnnotation.type === 'PromiseTypeAnnotation') {
-                  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
+                if (
+                  propertyTypeAnnotation.type === 'FunctionTypeAnnotation' ||
+                  propertyTypeAnnotation.type === 'PromiseTypeAnnotation' ||
+                  propertyTypeAnnotation.type === 'VoidTypeAnnotation'
+                ) {
+                  const invalidPropertyValueType =
+                    UnsupportedObjectPropertyTypeToInvalidPropertyValueTypeMap[
+                      propertyTypeAnnotation.type
+                    ];
+
+                  throwUnsupportedObjectPropertyValueTypeAnnotationParserError(
                     hasteModuleName,
                     property.typeAnnotation.typeAnnotation,
                     property.key,
-                    'Promise',
+                    invalidPropertyValueType,
                   );
+                } else {
+                  return {
+                    name: key.name,
+                    optional,
+                    typeAnnotation: wrapNullable(
+                      isPropertyNullable,
+                      propertyTypeAnnotation,
+                    ),
+                  };
                 }
```

Comment:
with the suggested changes, this code will just become:
```ts
throwIfPropertyValueTypeIsUnsupported(/*params*/)
return {
  name: key.name,
  optional,
  typeAnnotation: wrapNullable(
    isPropertyNullable,
    propertyTypeAnnotation,
  ),
};
```

Reply:
Hello @cipolleschi, I did the suggested changes but after doing them I get an error with flow as `typeAnnotation` expects a value of type `NativeModuleBaseTypeAnnotation` but the type returned by `wrapNullable` is of type `NativeModuleTypeAnnotation`.

It was working before due to flow's type refinement because of the if condition and throw statement. I looked into the Flow documentation but couldn’t find a solution to it. Do you have any suggestions on how I should proceed?

Reply:
That's... weird. :/ 
Let's keep the `else`, then..